utils/serde/src/main/java/io/apicurio/registry/utils/serde/JsonSchemaKafkaDeserializer.java

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright 2020 Red Hat
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.apicurio.registry.utils.serde;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+
+import javax.ws.rs.core.Response;
+
+import org.apache.kafka.common.header.Header;
+import org.apache.kafka.common.header.Headers;
+import org.apache.kafka.common.serialization.Deserializer;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.worldturner.medeia.api.StringSchemaSource;
+import com.worldturner.medeia.api.jackson.MedeiaJacksonApi;
+import com.worldturner.medeia.schema.validation.SchemaValidator;
+
+import io.apicurio.registry.client.RegistryService;
+import io.apicurio.registry.rest.beans.ArtifactMetaData;
+import io.apicurio.registry.rest.beans.VersionMetaData;
+import io.apicurio.registry.utils.serde.util.Utils;
+
+/**
+ * @author eric.wittmann@gmail.com
+ */
+public class JsonSchemaKafkaDeserializer<T> extends AbstractKafkaSerDe<JsonSchemaKafkaDeserializer<T>> implements Deserializer<T> {
+
+    public static final String REGISTRY_JSON_SCHEMA_DESERIALIZER_VALIDATION_ENABLED = "apicurio.registry.serdes.json-schema.validation-enabled";
+
+    private static MedeiaJacksonApi api = new MedeiaJacksonApi();
+    private static ObjectMapper mapper = new ObjectMapper();
+    
+    private boolean validationEnabled = false;
+    private SchemaCache<SchemaValidator> schemaCache;
+
+    /**
+     * Constructor.
+     */
+    public JsonSchemaKafkaDeserializer() {
+        this(null);
+    }
+
+    /**
+     * Constructor.
+     * @param client
+     */
+    public JsonSchemaKafkaDeserializer(RegistryService client) {
+        super(client);
+    }
+
+    /**
+     * @see org.apache.kafka.common.serialization.Deserializer#configure(java.util.Map, boolean)
+     */
+    @Override
+    public void configure(Map<String, ?> configs, boolean isKey) {
+        super.configure(configs);
+
+        Object ve = configs.get(REGISTRY_JSON_SCHEMA_DESERIALIZER_VALIDATION_ENABLED);
+        this.validationEnabled = Utils.isTrue(ve);
+
+        this.schemaCache = new SchemaCache<SchemaValidator>(getClient()) {
```

Comment:
Cache needs to be instantiated outside configure() ... as this method might not be called -- when this deserializer is instantiated outside Kafka code.

Reply:
Ah ok - that goes for validationEnabled too then.

Reply:
Yeah ... I would make validationEnabled a Boolean (non-primitive), add a setter, and if it was explicitly set (not-null), I wouldn't handle it in configure()